File: tv/src/main/java/com/synergics/stb/iptv/leanback/models/db/TVCategoriesDao.java
package com.synergics.stb.iptv.leanback.models.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.synergics.stb.iptv.leanback.models.TVCategories;

import java.util.List;
@Dao
public interface TVCategoriesDao {
    @Query("SELECT * FROM tv_categories")
    List<TVCategories> all();

    @Query("SELECT * FROM tv_categories WHERE nama=:nama")
    TVCategories getByNama(String nama);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(TVCategories categories);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long[] insert(TVCategories... categories);

    @Update
    void update(TVCategories... categories);

    @Delete
    void delete(TVCategories... categories);
}

File: tv/src/main/java/com/synergics/stb/iptv/leanback/models/db/TVCategoriesDao.kt
package com.synergics.stb.iptv.leanback.models.db

import androidx.room.*
import com.synergics.stb.iptv.leanback.models.TVCategories

@Dao
interface TVCategoriesDao {
    @Query("SELECT * FROM tv_categories")
    fun all(): List<TVCategories?>?

    @Query("SELECT * FROM tv_categories WHERE nama=:nama")
    fun getByNama(nama: String?): TVCategories?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(categories: TVCategories?): Long

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg categories: TVCategories?): LongArray?

    @Update
    fun update(vararg categories: TVCategories?)

    @Delete
    fun delete(vararg categories: TVCategories?)

    @Query("delete from tv_categories")
    fun clear()

    @Query("delete from sqlite_sequence where name='tv_categories'")
    fun resetAI()
}